src/Metadata/Property/Factory/AnnotationPropertyNameCollectionFactory.php

```diff
@@ -63,7 +63,7 @@ public function create(string $resourceClass, array $options = []): PropertyName
         // Properties
         foreach ($reflectionClass->getProperties() as $reflectionProperty) {
             if (null !== $this->reader->getPropertyAnnotation($reflectionProperty, ApiProperty::class)) {
-                $propertyNames[$reflectionProperty->name] = true;
+                $propertyNames[$reflectionProperty->name] = (string)$reflectionProperty->name;
```

Comment:
`$reflectionProperty->name` is a always a string, so the cast isn't necessary (same everywhere).